examples/02_decoding/plot_miyawaki_encoding.py

```diff
@@ -0,0 +1,259 @@
+plt.axis('off')
+plt.title('Run {}, Stimulus {}'.format(3, 102))
+plt.subplots_adjust(wspace=0.5)
+
+##############################################################################
+# We now stack the X and y data and remove an offset in the beginning/end.
+
+X_train = np.vstack([x[2:] for x in X_train])
+y_train = np.vstack([y[:-2] for y in y_train]).astype(float)
+
+##############################################################################
+# X_train is a matrix of *samples* x *voxels*
+
+print(X_train.shape)
+
+##############################################################################
+# We flatten the last two dimensions of y_train
+# so it is a matrix of *samples* x *pixels*.
+
+# Flatten the stimuli
+y_train = np.reshape(y_train, (-1, y_shape[0] * y_shape[1]))
+
+print(y_train.shape)
+
+##############################################################################
+# Building the encoding models
+# ----------------------------
+# We can now proceed to build a simple **voxel-wise encoding model** using
+# `Ridge regression <http://en.wikipedia.org/wiki/Tikhonov_regularization>`_.
+# For each voxel we fit an independent regression model,
+# using the pixel-values of the visual stimuli to predict the neuronal
+# activity in this voxel.
+
+from sklearn.linear_model import Ridge
+from sklearn.cross_validation import KFold
+
+##############################################################################
+# Using 10-fold cross-validation, we partition the data into 10 'folds',
+# we hold out each fold of the data for testing, then fit a ridge regression
+# to the remaining 9/10 of the data, using y_train as predictors
+# and X_train as targets, and create predictions for the held-out 10th.
+from sklearn.metrics import r2_score
+
+estimator = Ridge(alpha=100.)
+cv = KFold(len(y_train), 10)
+
+scores = []
+for train, test in cv:
+    # we train the Ridge estimator on the training set
+    # and predict the fMRI activity for the test set
+    predictions = Ridge(alpha=100.).fit(
+    y_train.reshape(-1, 100)[train], X_train[train]).predict(
+        y_train.reshape(-1, 100)[test])
+    # we compute how much variance our encoding model explains in each voxel
+    scores.append(r2_score(X_train[test], predictions,
+                           multioutput='raw_values'))
+
+##############################################################################
+# Mapping the encoding scores on the brain
+# ----------------------------------------
+# To plot the scores onto the brain, we use :func:`nilearn.image.new_img_like`
+# to create a Nifti1Image containing the scores and then threshold it:
+
+from nilearn.image import threshold_img, new_img_like
+from nilearn.masking import unmask
+
+cut_score = np.array(scores).mean(axis=0)
+cut_score[cut_score < 0] = 0
+
+# bring the scores into the shape of the background brain
+score_map_img = new_img_like(dataset.background,
+                         unmask(cut_score, dataset.mask).get_data())
```

Comment:
We tend to prefer using masker.inverse_transform. Is there a reason why it is not suitable here?